File: core/query_language/semantic_query_optimizer.py
from __future__ import annotations

from typing import Any, Dict


def optimize_semantic_query(
    plan: Dict[str, Any],
) -> Dict[str, Any]:

    optimized_steps = sorted(
        plan.get("steps", []),
        key=lambda x: x.get("operation", ""),
    )

    return {
        "steps": optimized_steps,
        "limit": plan.get("limit", 100),
        "optimized": True,
    }

File: src/query_language/semanticQueryOptimizer.ts
/**
 * Converted from Python: core/query_language/semantic_query_optimizer.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function optimizeSemanticQuery(plan: any): any {
  var optimized_steps: any = py.sorted(py.get(plan, "steps", []), {key: ((x: any) => py.get(x, "operation", "")) as (item: any) => any});
  return {"steps": optimized_steps, "limit": py.get(plan, "limit", 100), "optimized": true};
}
